src/Service/Daemon.hs

```diff
@@ -513,8 +524,8 @@ run' threadMapTV = do
                      M.filterWithKey (\thisAutomationName _action -> thisAutomationName /= automationName) actions
                  in
                    ( -- if the topic map has become empty as a
-                     -- result of this operation
-                     if not $ null (M.keys actions) && M.null updatedActions then
```

Comment:
This is the bug, this line was the problem. Not sure where my brain was at...but more problematically, this wasn't tested in any way.